Sources/CreateOptions/PackageDeclaration.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+
+public enum DependencyRule: Decodable {
+    case exact(version: String)
+    case upToNextMajor(from: String)
+    case upToNextMinor(from: String)
+    case range(from: String, to: String)
+    case closedRange(from: String, to: String)
+    case branch(name: String)
+    case commit(hash: String)
+    case from(version: String)
+
+    var declaration: String {
+        switch self {
+        case .exact(version: let version):
+            return ".exact(\"\(version)\")"
+        case .upToNextMajor(from: let version):
+            return ".upToNextMajor(from: \"\(version)\")"
+        case .upToNextMinor(from: let version):
+            return ".upToNextMinor(from: \"\(version)\")"
+        case .range(from: let min, to: let max):
+            return "\"\(min)\"..<\"\(max)\""
+        case .closedRange(from: let min, to: let max):
+            return "\"\(min)\"...\"\(max)\""
+        case .branch(name: let name):
+            return ".branch(\"\(name)\")"
+        case .commit(hash: let hash):
+            return ".revision(\"\(hash)\")"
+        case .from(let version):
+            return "from: \"\(version)\""
+        }
+    }
+}
+
+public struct PackageDeclaration: Decodable {
+    
+    let url: URL
+    let module: String
```

Comment:
I'm not sure what is the most appropriate term to use here. After reading through https://forums.swift.org/t/module-vs-product-vs-target/27640, I have a feeling that what we actually want is 'product'.

Additionally, maybe this needs to be an array in scenarios where you need multiple products offered by the package? 

For example:

```yaml
package:
  dependencies:
  - url: https//github.com/apple/swift-syntax
    from: 0.50600.1
    products:
    - SwiftSyntax
    - SwiftSyntaxParser

Reply:
I actually did use an array but with testing, but by my testing when a package has dependencies on another package it imports all of the products? I don't know the specifics.

By my testing, I had SwiftNumerics as a dependency as only had Numerics as a product. I imported the generated package and was able to use RealNumerics and ComplexModule.

Reply:
My testing from within the package is consistent with what I tested before with an external project, we should only have to label one product and all other products are imported as well.

Reply:
It was probably cached in derived data/the build directory. Try cleaning the project and it won't work (or `swift build -c release`), you have to specify all products that are required so that it knows what to link.

Reply:
Apologies, I got confused because a target will also import its dependencies and `Numerics` imported `RealNumerics` and `ComplexModule` as well.

Reply:
Do we think we can generalize `OptionSetDecodable` so that the wrapped type doesn't have to be `CaseIterable`? It would help with the array syntax at least a bit as I assume most imports will probably use single products.

Reply:
I don't think we can so much generalize it (or at least I didn't figure out a way to), but we could just add a similar `SingleElementDecodable` property wrapper instead?